gradle-android-command-plugin/src/main/groovy/com/novoda/gradle/command/Monkey.groovy

```diff
@@ -15,12 +16,29 @@ class Monkey extends AdbTask {
 
     @TaskAction
     void exec() {
-        def arguments = ['shell', 'monkey', '-p', packageName, '-v', getEvents()]
+        def arguments = ['shell', 'monkey', '-p', packageName]
+        arguments += getFormattedCategories()
+        arguments += ['-v', getEvents()]
         if (getSeed())
             arguments += ['-s', getSeed()]
         assertDeviceAndRunCommand(arguments)
     }
 
+    def getFormattedCategories() {
+        def formattedCategories = []
+        def categories = getCategories()
+        categories.each {
+            formattedCategories.add("-c " + it)
+        }
+        formattedCategories
+    }
```

Comment:
Seems very Java-ish. Can't you just transform getCategories() with the collect function and a closure? Think this would be a simple single line function...